fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/DimensionsApiRequestImpl.java

```diff
@@ -167,7 +167,7 @@ private DimensionsApiRequestImpl(
         for (String apiFilter : apiFilters) {
             ApiFilter newFilter;
             try {
-                newFilter = ApiFilterGenerator.build(apiFilter, dimensionDictionary);
+                newFilter = generateFilter(apiFilter, dimensionDictionary);
```

Comment:
I'm confused here. It looks like in the DataApiRequestImpl, we're using the
FilterGenerator to parse the *entire* filterString in one shot. However, in
the DimensionsApiRequestImpl, we're first splitting the filters and then passing
each one in one at a time. Why the difference? Why can't we just pass in the
`filterQuery` string to the filter generator in one shot?

Reply:
This is a case where two different refactors were done at different times.  I'll align them.

Reply:
Nope.  The filter builder in DimensionApiRequest doesn't bind using a logical table.  The one in DataApiRequest does.

Reply:
Hmm. Ok, I'm willing to pass on this for now, then. It'd be nice if we could find some way of aligning the interfaces for generating filters (reduces cognitive overhead in my opinion), but it's not worth thinking more deeply about now.